app/Http/Controllers/AdvertApplicationController.php

```diff
@@ -89,33 +102,59 @@ public function store(Advert $job, StoreAdvertApplicationRequest $request)
 
     /**
      * Display the specified resource.
+     *
+     * @param App\Models\Application $application
+     * @return \Illuminate\Http\Response
      */
     public function show(Application $application): Response
     {
-        abort(404);
+        $response = Gate::inspect('view', $application);
+
+        if ($response->denied()) {
+            abort(404);
+        }
+
+        Log::info("Showing application #$application->id");
+
+        return response()->view('jobs.application', ['application' => $application]);
     }
 
     /**
      * Show the form for editing the specified resource.
+     *
+     * @param App\Models\Application $application
+     * @return \Illuminate\Http\Response
      */
     public function edit(Application $application): Response
     {
-        abort(404);
+        return response()->view('applications.edit', ['application' => $application]);
     }
 
     /**
      * Update the specified resource in storage.
+     *
+     * @param App\Http\Requests\UpdateJobApplicationRequest $request
+     * @param App\Models\Application $application
+     * @return \Illuminate\Http\Response
      */
     public function update(UpdateAdvertApplicationRequest $request, Application $application): Response
     {
-        abort(404);
+        $request->validated();
+
+        $application->update([
+            'content' => $request->content,
+        ]);
+
+        return response()->view('/');
     }
 
     /**
      * Remove the specified resource from storage.
      */
     public function destroy(Application $application): Response
```

Comment:
Remove the `: Response` type hint and add `@return \Illuminate\Http\Response` as a PHPdoc comment